diff --git a/src/freedreno/ir3/ir3_compiler.c b/src/freedreno/ir3/ir3_compiler.c
index 2f35c19f235..2ca0019e532 100644
--- a/src/freedreno/ir3/ir3_compiler.c
+++ b/src/freedreno/ir3/ir3_compiler.c
@@ -203,7 +203,7 @@ static const nir_shader_compiler_options ir3_base_options = {
 static void
 __debug_init(void)
 {
-   ir3_shader_debug = debug_get_option_ir3_shader_debug();
+   ir3_shader_debug = debug_get_option_ir3_shader_debug() | IR3_DBG_NOFP16;
    ir3_shader_override_path =
       __normal_user() ? debug_get_option_ir3_shader_override_path() : NULL;
 
diff --git a/src/freedreno/vulkan/tu_util.cc b/src/freedreno/vulkan/tu_util.cc
index 01c7102b52c..e4fab7279d0 100644
--- a/src/freedreno/vulkan/tu_util.cc
+++ b/src/freedreno/vulkan/tu_util.cc
@@ -129,7 +129,7 @@ tu_env_deinit(void)
 static void
 tu_env_init_once(void)
 {
-   tu_env.start_debug = tu_env.debug = parse_debug_string(os_get_option("TU_DEBUG"), tu_debug_options);
+   tu_env.start_debug = tu_env.debug = parse_debug_string(os_get_option("TU_DEBUG"), tu_debug_options) | TU_DEBUG_NOUBWC;
 
    if (TU_DEBUG(STARTUP))
       mesa_logi("TU_DEBUG=0x%" PRIx64, tu_env.debug.load());
